CMakeLists.txt

```diff
@@ -41,6 +41,12 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
 elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
     target_compile_options(gfx PRIVATE $<$<COMPILE_LANGUAGE:CXX>:/W4 /WX /experimental:external /external:anglebrackets /external:W0 /analyze:external->)
     target_compile_options(gfx PRIVATE $<$<COMPILE_LANGUAGE:CXX>:/wd4324>) # structure was padded due to alignment specifier
+    target_compile_options(gfx PRIVATE $<$<COMPILE_LANGUAGE:CXX>:/wd6387>) # param could be 0
+    target_compile_options(gfx PRIVATE $<$<COMPILE_LANGUAGE:CXX>:/wd6262>) # Function uses alot of stack
+    target_compile_options(gfx PRIVATE $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<CONFIG:Debug>>:/wd6011>) # Dereferencing NULL pointer
+    target_compile_options(gfx PRIVATE $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<CONFIG:Debug>>:/wd28183>) # Variable could be '0'
+    target_compile_options(gfx PRIVATE $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<CONFIG:Debug>>:/wd6385>) # Reading invalid data
+    target_compile_options(gfx PRIVATE $<$<AND:$<COMPILE_LANGUAGE:CXX>,$<CONFIG:Debug>>:/wd6386>) # Buffer overrun while writing
```

Comment:
Unfortunately the gfx aseert macro causes errors to be thrown that otherwise wouldnt have as such Ive bypassed them on debug builds

Reply:
Alternatively I suppose we could include some "#pragma warning push/pop" at the start and end of the `GFX_ASSERT` macro.
But I'm ok with this CMake-based approach too.

Reply:
Unfortunately the errors arent actually in the assert macro but instead because of the presence of checks for whatever is in the assert statement, which causes the analysis to function differently and throw an error it wouldnt otherwise throw.
So as a simple case: if we have a pointer that is flagged as not null (using the gfxMalloc workaround) then there shouldnt be any errors about de-referencing null pointers however if we put a `GFX_ASSERT(pointer != NULL)` somewhere then the analysis will now assume that the pointer can in fact now be null (as otherwise why would you be checking for it) and throw an error. We could workaround this by having assert actually terminate on error as then we would actually be handling the error in some way, wheras currently we just print (and only if a debugger is attached) and continue